### ISSUE TITLE
Info diagnostic for crashing diagnostics missing from error list sometimes

### ISSUE THREAD

**Issue by shyamnamboodiripad** [dotnet/roslyn#1521]
I created a solution with two analyzer projects one VB and one C# (both created using Roslyn SDK project templates). In the VB one, I added code to throw NRE from the SupportedDiagnostics property. In the C# one, I added code to throw NRE from Initialize() method. I then built the solution, set one of the two VSIX projects as startup project and hit Ctrl + F5.

In the resulting instance of VS, I opened a solution containing one VB console app and one C# console app. My expectation was that I should see two info diagnostics in the error list one for the crash from the VB analyzer and another for the crash from the C# one.

But sometimes, I see only the info diagnostic for the C# exception when I open the solution and at other times I see only the one for the VB exception. Looks like there is some problem in the way these exceptions are reported.

Example where only VB exception is reported:
![capture](https://cloud.githubusercontent.com/assets/10579684/6798617/27d84004-d1cc-11e4-97b3-85166fd811c5.PNG)

Example where only C# exception is reported:
![capture2](https://cloud.githubusercontent.com/assets/10579684/6798709/237b2bd8-d1cd-11e4-904a-26f107a18433.PNG)

**Comment by mavasani** [comment]
@shyamnamboodiripad did you open source files from both projects in both repros? we only try to execute analyzers specific to project language, so you may have to do so or wait little longer with closed file diagnostic turned on.

**Comment by shyamnamboodiripad** [comment]
Yes iirc, I had both language files open. As you can see from the above screenshot, I was seeing different info diagnostics for the same solution when I opened it repeatedly (with the same set of files opened - I didn't close source files between the tries as far as I can recall).

**Comment by mavasani** [comment]
@shyamnamboodiripad I was able to get a consistent repro for this:

1. I created a solution with two analyzer projects one VB and one C# (both created using Roslyn SDK project templates). In the VB one, I added code to throw NRE from the SupportedDiagnostics property. In the C# one, I added code to throw NRE from Initialize() method. I then built the solution, set one of the two VSIX projects as startup project and hit Ctrl + F5.

2. In the resulting instance of VS, create a new C# project and observe the analyzer exception diagnostic in error list.

3. Add a VB project to this solution and observe second analyzer exception diagnostic in error list.

4. Close the solution and create a new C# project (new solution, say Sln2) and observe the analyzer exception diagnostics in error list. Note that even the analyzer exception diagnostic from VB analyzer shows up. This is expected as the VB analyzer was also a host session wide analyzer, whose diagnostic surives entire VS session.

5. Now close VS and reopen a new VS instance with the same analyzers. In this instance re-open Sln2 with the single C# project.

Excepted: Analyzer exception diagnostic shows up in error list
Actual: No diagnostic in error list.

When I attached a debugger, I see that analyzer exception diagnostic is in fact generated and reported to AbstractHostDiagnosticUpdateSource, but error list event handler isn't hooked up to DiagnosticSource yet, causing this diagnostic not to reach the error list.
Another interesting fact is that if I delete the ".vs" directory for the C# project in Sln2 and repeat step 5. then the analyzer exception diagnostic does indeed show up in error list.
This probably seems related to #1540. Anycase I think @heejaechang is the best person to take a look at this.
